src/app/modules/core/components/user-profile/contact-list/contact-list.component.ts

```diff
@@ -0,0 +1,78 @@
+import { AfterViewInit, Component, OnInit, ViewChild } from '@angular/core';
+import { MatPaginator } from '@angular/material/paginator';
+import { MatSort } from '@angular/material/sort';
+import { MatTableDataSource } from '@angular/material/table';
+import {
+  COMPOSITE_FIELD,
+  DATE_OF_BIRTH,
+  LOCATION,
+  NAMES,
+  PAGE_SIZE_OPTIONS,
+  PROFILE_PIC,
+} from '../../../constants/appConstants';
+import { UserData } from '../../../interfaces/user-data';
+
+@Component({
+  selector: 'app-contact-list',
+  templateUrl: 'contact-list.component.html',
+  styleUrls: ['contact-list.component.css'],
+})
+export class ContactListComponent implements  AfterViewInit {
+  public displayedColumns: string[];
+  public dataSource: MatTableDataSource<UserData>;
+  public pageSizeOptions = PAGE_SIZE_OPTIONS;
+
+  @ViewChild(MatPaginator) paginator: MatPaginator;
+  @ViewChild(MatSort) sort: MatSort;
+
+  constructor() {
+    const self = this;
+    this.displayedColumns = [
+      'id',
+      'profilePic',
+      'name',
+      'compositeField',
+      'dateOfBirth',
+      'location',
+    ];
+    const users = Array.from({ length: 50 }, (_, k) =>
+      self.createNewUser(k + 1)
```

Comment:
In real world scenario, this would be an API call. So we can create a parser for User array. The functionality of parser would be to process input we are receiving from server and add/delete any extra information if needed. Say for eg:- We are getting { first name, last name, address, phone number, prof pic, etc } from server. We want to show "FULL NAME" instead of first name and last name, also we want to show contact info which includes both address and phone number in our table. In that case, our parser will iterate over user object array, returns required object after processing. In this case, we dont have to provide extra string array for column names, our html should be able to get column names from object key.